src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputParameterConverter.cs

```diff
@@ -0,0 +1,124 @@
+            {
+                var isKnownProperty = reader.TryReadReferenceId(ref isFirstProperty, ref id)
+                    || reader.TryReadString(nameof(InputParameter.Name), ref name)
+                    || reader.TryReadString(nameof(InputParameter.NameInRequest), ref nameInRequest)
+                    || reader.TryReadString(nameof(InputParameter.Description), ref description)
+                    || reader.TryReadWithConverter(nameof(InputParameter.Type), options, ref parameterType)
+                    || reader.TryReadString(nameof(InputParameter.Location), ref location)
+                    || reader.TryReadWithConverter(nameof(InputParameter.DefaultValue), options, ref defaultValue)
+                    || reader.TryReadWithConverter(nameof(InputParameter.VirtualParameter), options, ref virtualParameter)
+                    || reader.TryReadWithConverter(nameof(InputParameter.GroupedBy), options, ref groupBy)
+                    || reader.TryReadString(nameof(InputParameter.Kind), ref kind)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsRequired), ref isRequired)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsApiVersion), ref isApiVersion)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsResourceParameter), ref isResourceParameter)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsContentType), ref isContentType)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsEndpoint), ref isEndpoint)
+                    || reader.TryReadBoolean(nameof(InputParameter.SkipUrlEncoding), ref skipUrlEncoding)
+                    || reader.TryReadBoolean(nameof(InputParameter.Explode), ref explode)
+                    || reader.TryReadString(nameof(InputParameter.ArraySerializationDelimiter), ref arraySerializationDelimiter)
+                    || reader.TryReadString(nameof(InputParameter.HeaderCollectionPrefix), ref headerCollectionPrefix);
+
+                if (!isKnownProperty)
+                {
+                    reader.SkipProperty();
+                }
+            }
+
+            name = name ?? throw new JsonException("Parameter must have name");
+            nameInRequest = nameInRequest ?? throw new JsonException("Parameter must have nameInRequest");
+            parameterType = parameterType ?? throw new JsonException("Parameter must have type");
+
+            if (location == null)
+            {
+                throw new JsonException("Parameter must have location");
+            }
+            RequestLocation requestLocation;
+            Enum.TryParse<RequestLocation>(location, ignoreCase: true, out requestLocation);
+
+            if (kind == null)
+            {
+                throw new JsonException("Parameter must have kind");
+            }
+            InputOperationParameterKind parameterKind;
+            Enum.TryParse<InputOperationParameterKind>(kind, ignoreCase: true, out parameterKind);
```

Comment:
How about we combine these two lines or just use Enum.Parse and let it panic when it cannot 
```suggestion
            Enum.TryParse<InputOperationParameterKind>(kind, ignoreCase: true, out var parameterKind);
```